### ISSUE TITLE
provider name detection can be wrong when using symlinks

### ISSUE THREAD

**Issue by mingfang** [dikhan/terraform-provider-openapi#46]
The use of os.Executable() can be wrong when the plugin is a symlink, ```ln -sF terraform-provider-openapi terraform-provider-<your_provider_name>``` may yield a name of ```openapi``` which is wrong.

 https://github.com/dikhan/terraform-provider-openapi/blob/03d1765264c7e08bdef63534651473fa5c64def8/openapi/provider.go#L67

**Comment by dikhan** [comment]
closed this issue by mistake, awaiting @mingfang to provide more details about symlink issue

**Comment by dikhan** [comment]
hi @mingfang quickly following up with this...can you pls provide an example on the failing use case you mentioned? Also, what OS are you using?

Thanks,
Dani

**Comment by mingfang** [comment]
The golfing doc says this can happen.
https://golang.org/pkg/os/#Executable

I'm doing a custom build inside Docker and can workaround this.  
If it doesn't affect you then please feel free to close this ticket.

**Comment by dikhan** [comment]
+1, symlinks support was added to the install script as well as the make files to optimize on memory when installing multiple openapi providers that use the same underlying open api provider code base...but as pointed out by @mingfang and based on the documentation there is no guarantee that the path is still pointing to the correct executable (if a symlink was used to start the process, depending on the operating system, the result might be the symlink or the path it pointed to). Hence, I decided to remove symlinks from the repo for now until it becomes an actual need. For now, the executable call will work just fine as the expectation is for the binaries to be executables and not symlinks.

If symlinks are needed in the future, let's discuss about adding support for that then.

@mingfang FYI: I have added a new target to the make file called ```examples-container``` that brings up all the example APIs and also starts a container set up with the example openapi terraform providers so users can play around with different terraform configurations using openapi resources in an environment that is already set up with terraform and the example openapi terraform providers (goa and swaggercodegen).

I will go ahead and close this ticket as corresponding work towards removing symlinks references has already been done as part of issue #62 

Thanks,
Dani
